### PR TITLE
Issue #13321: Kill mutation for SummaryJavaDocCheck-3

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13321: Kill mutation for SummaryJavaDocCheck-3

-----

# Check :- 
https://checkstyle.org/checks/javadoc/summaryjavadoc.html#SummaryJavadoc

-----

# Mutation 
https://github.com/checkstyle/checkstyle/blob/ce6a1113b574960e8b7e311af4c279aad24ed71b/config/pitest-suppressions/pitest-javadoc-suppressions.xml#L885-L901

-----

# Explaination
getSummaryDentence is used only at `validateUntaggedSummary` 

Removal of https://github.com/checkstyle/checkstyle/blob/ce6a1113b574960e8b7e311af4c279aad24ed71b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java#L674-L676 will not make any issue the loop run 1 time more.

https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.html#JAVADOC

according to this instead of loop stop at javadoc tag ot will stop at eof. and it doesn't effect output.

lets assume in case if we have code like this
```
    /** 
     *  Comment of java.
     *  
     *  @param a
     *  worst case test
     */
    public void foo1(int a) throws Exception {
    }
```

then `worst case test` become descripion which is also under javadoctag
```
    |   |   |   |       |--WS ->    [18:6]
    |   |   |   |       |--JAVADOC_TAG -> JAVADOC_TAG [18:8]
    |   |   |   |       |   |--PARAM_LITERAL -> @param [18:8]
    |   |   |   |       |   |--WS ->   [18:14]
    |   |   |   |       |   |--PARAMETER_NAME -> a [18:15]
    |   |   |   |       |   |--NEWLINE -> \n [18:16]
    |   |   |   |       |   `--DESCRIPTION -> DESCRIPTION [19:0]
    |   |   |   |       |       |--LEADING_ASTERISK ->      * [19:0]
    |   |   |   |       |       |--TEXT ->   worst case test [19:6]
    |   |   |   |       |       |--NEWLINE -> \n [19:23]
    |   |   |   |       |       `--TEXT ->       [20:0]
    |   |   |   |       `--EOF -> <EOF> [20:5]

```

-----

# Regression :- 

Report-1 :- https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/58c7e7f_2023162747/reports/diff/index.html

Report-2 :- https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/58c7e7f_2023105658/reports/diff/index.html

-----

Diff Regression config: https://gist.githubusercontent.com/Kevin222004/61eaa69fe98585b6ee1882ca082362bc/raw/316ca02ce9fd290119d9bfc560f2b11e84a1b685/Summary.xml
Diff Regression projects: https://gist.githubusercontent.com/Kevin222004/21e3934e85f802e2fbd48af06d122364/raw/604256badd733d8568064f371d55657c04b00dfd/test-projects-2.properties
Report label: change-config-Regression-1

### CODE REVIEW

**Comment by Kevin222004** [comment]
Github, generate report

**Comment by Kevin222004** [comment]
Github, generate report

**Comment by Kevin222004** [comment]
Github, generate report

**Comment by Kevin222004** [comment]
Github, generate report

File: config/pitest-suppressions/pitest-javadoc-suppressions.xml [hunk 1]
```diff
@@ -882,23 +882,25 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (rdiachenko) [inline config/pitest-suppressions/pitest-javadoc-suppressions.xml:903]: What's the reason for empty lines?

Comment (romani) [reply config/pitest-suppressions/pitest-javadoc-suppressions.xml:903]: Some sort of avoiding conflicts

**Review by rdiachenko — APPROVED** [review]
lgtm

**Review by romani — APPROVED** [review]
Ok to merge

**Review by Vyom-Yadav — APPROVED** [review]
LGTM!

**Pull request merged**